allennlp/common/util.py

```diff
@@ -62,6 +64,27 @@
 ContextManagerFunctionReturnType = Generator[T, None, None]
 
 
+class PretrainedTransformer:
```

Comment:
I know this wasn't changed in this PR, but I think having this class is a _super_ bad idea - it has reallly gross side effects if you modify the model, e.g setting `requires_grad`. We should just eat the cost of loading the model twice. It takes < 10 seconds and uses max ~2GB memory.

Reply:
Yeah, fair enough, there aren't a lot of places where we'd actually want this caching.  I just removed it.